client/src/helpers/hooks/usePagination.ts

```diff
@@ -0,0 +1,94 @@
+import { useMemo } from 'react';
+
```

Comment:
I'll  hold off on in depth review until logic is implemented - looks good so far :)